GeeksCoreLibrary/Modules/Templates/Services/LegacyTemplatesService.cs

```diff
@@ -1275,6 +1278,13 @@ public async Task<TemplateDataModel> GetTemplateDataAsync(ITemplatesService temp
             foreach (var templateId in template.CssTemplates.Concat(template.JavascriptTemplates))
             {
                 var linkedTemplate = await templatesService.GetTemplateAsync(templateId);
+
+                // Validate the template regex, if it has one.
+                if (!String.IsNullOrWhiteSpace(linkedTemplate.UrlRegex) && !String.IsNullOrWhiteSpace(currentUrl) && !Regex.IsMatch(currentUrl, linkedTemplate.UrlRegex))
```

Comment:
Kun je aub een timeout toevoegen aan de `Regex.IsMatch`, zoals ik laatst had aangegeven in .NET overleg?

Reply:
Ik heb een timeout toegevoegd.